power-sequencer/mihawk-cpld.cpp

```diff
@@ -217,6 +221,81 @@ void MIHAWKCPLD::onFailure()
 
 void MIHAWKCPLD::analyze()
 {
+    //analyze psu status when power_on fault.
+    auto poweronError = checkPoweronFault();
+    if(poweronError)
+    {
+        int errorcode;
+        errorcode = checkPSUDCpgood(true);
+        if (!((errorcode >> 1) & 0) && !((errorcode >> 3) & 0) )
```

Comment:
ANDing something with zero will always be zero, so I don't think this will work quite right.

Is this an equivalent?  if (errorcode & 0xA)
which checks for bits 0x1010

Reply:
@spinler   For PSU1, this part needs to check whether the first bit and the third bit are "0" at the same time, if both are 0, it means PSU1 power fail.
For PSU0, needs to check whether the second bit and the fourth bit are "0" at the same time.

I will revise this part